nerfstudio/data/datamanagers/full_images_datamanager.py

```diff
@@ -172,14 +172,31 @@ def cache_images(self, cache_images_option):
             if mask is not None:
                 data["mask"] = mask
 
-            cached_eval.append(data)
+            self.eval_dataset.cameras.fx[idx] = float(K[0, 0])
+            self.eval_dataset.cameras.fy[idx] = float(K[1, 1])
+            self.eval_dataset.cameras.cx[idx] = float(K[0, 2])
+            self.eval_dataset.cameras.cy[idx] = float(K[1, 2])
+            self.eval_dataset.cameras.width[idx] = image.shape[1]
+            self.eval_dataset.cameras.height[idx] = image.shape[0]
+            return data
 
-            self.eval_dataset.cameras.fx[i] = float(K[0, 0])
-            self.eval_dataset.cameras.fy[i] = float(K[1, 1])
-            self.eval_dataset.cameras.cx[i] = float(K[0, 2])
-            self.eval_dataset.cameras.cy[i] = float(K[1, 2])
-            self.eval_dataset.cameras.width[i] = image.shape[1]
-            self.eval_dataset.cameras.height[i] = image.shape[0]
+        result_list = []
+        CONSOLE.log("Caching / undistorting train images")
+        with ThreadPoolExecutor(max_workers=multiprocessing.cpu_count()) as executor:
+            for idx in range(len(self.train_dataset)):
+                result = executor.submit(process_train_data, idx)
+                result_list.append(result)
+            for data in track(result_list, description="Caching / undistorting train images", transient=True):
+                cached_train.append(data.result())
```

Comment:
Untested, but I think you can replace all of these loops with something like:

```python
cached_train = list(
    executor.map(process_train_data, track(range(len(self.train_dataset))))
)
```

Reply:
Yes, this works with a faster speed. Thank you!